src/dmd/utils.d

```diff
@@ -91,20 +91,25 @@ extern (C++) void writeFile(Loc loc, File* f)
  *   loc = The line number information from where the call originates
  *   name = a path to check (the name is stripped)
  */
-extern (C++) void ensurePathToNameExists(Loc loc, const(char)* name)
+void ensurePathToNameExists(Loc loc, const(char)[] name)
 {
-    const(char)* pt = FileName.path(name);
-    if (*pt)
+    const char[] pt = FileName.path(name);
+    if (pt.length)
     {
         if (!FileName.ensurePathExists(pt))
         {
-            error(loc, "cannot create directory %s", pt);
+            error(loc, "cannot create directory %s", pt.ptr);
```

Comment:
These changes are shot through with the assumption of 0 termination. I think these are subtle bugs waiting to trip up the next person modifying the code.

Reply:
The methods in filename all 0-terminate their returns.  I believe the only case where the slice isn't guaranteed to come from a filename function is the call to mkdir above.  I could use %*.s with the explicit length here instead to future-proof it though